### PR TITLE
refine coding styles for cpu&memory plugin

### PR DESCRIPTION
**Pull request by waynepeking348**
#### What type of PR is this?
refactor

#### What this PR does / why we need it:

currently, qrm plugin is a little complicated and lack of abstraction for similar functionalities

this pr is for cpu and memory plugin

### CODE REVIEW

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_test.go [hunk 1]
```diff
@@ -28,23 +28,24 @@ import (
...
+	"github.com/kubewharf/katalyst-core/pkg/metaserver/agent"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver/agent/pod"
 	"github.com/kubewharf/katalyst-core/pkg/metrics"
 	"github.com/kubewharf/katalyst-core/pkg/util/machine"
 )
 
 func getTestDynamicPolicyWithInitialization(topology *machine.CPUTopology, machineInfo *info.MachineInfo, stateFileDirectory string) (*DynamicPolicy, error) {
-	reservedMemory, err := getReservedMemory(4, machineInfo)
-
+	reservedMemory, err := util.GetReservedMemory(machineInfo, 4)
```
Comment (csfldf) [inline pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_test.go:48]: not common util

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go [hunk 2]
```diff
@@ -62,33 +64,32 @@ func (p *DynamicPolicy) dedicatedCoresHintHandler(ctx context.Context,
...
 	}
 
 	resourcesMachineState := p.state.GetMachineState()
-
 	var hints map[string]*pluginapi.ListOfTopologyHints
 
 	allocationInfo := p.state.GetAllocationInfo(v1.ResourceMemory, req.PodUid, req.ContainerName)
 	if allocationInfo != nil {
-		hints = regenerateHints(allocationInfo, uint64(reqInt))
+		hints = rePackHints(uint64(reqInt), allocationInfo)
```
Comment (csfldf) [inline pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go:88]: whar does re means

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy.go [hunk 3]
```diff
@@ -235,27 +235,27 @@ func (p *DynamicPolicy) GetTopologyHints(ctx context.Context, req *pluginapi.Res
 
 	p.RLock()
 	defer func() {
-		p.RUnlock()
-
 		if err != nil {
 			_ = p.emitter.StoreInt64(util.MetricNameGetTopologyHintsFailed, 1, metrics.MetricTypeNameRaw)
 		}
+		p.RUnlock()
```
Comment (csfldf) [inline pkg/agent/qrm-plugins/memory/dynamicpolicy/policy.go:241]: doesn't get it

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go [hunk 4]
```diff
@@ -97,27 +98,28 @@ func (p *DynamicPolicy) dedicatedCoresWithNUMABindingHintHandler(ctx context.Con
 			}
 
 			var err error
-			resourcesMachineState, err = state.GenerateResourcesMachineStateFromPodEntries(p.state.GetMachineInfo(), podResourceEntries, p.state.GetReservedMemory())
+			resourcesMachineState, err = state.GenerateMachineStateFromPodEntries(p.state.GetMachineInfo(), podResourceEntries, p.state.GetReservedMemory())
```
Comment (csfldf) [inline pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go:101]: hope not to modify it

**Comment by csfldf** [comment]
To be honest, I think the risk and burden of making such a large change to these codes are far more than the benefits, especially when it's already running online. But it's been done, I've tried my best to check if modifications will cause problems , but it's really hard to identify the difference between all the changes and the original codes.

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 5]
```diff
@@ -435,10 +439,9 @@ func (p *DynamicPolicy) GetResourcesAllocation(_ context.Context,
 	}
 
 	if len(allocationInfosJustFinishRampUp) > 0 {
-		err = p.putContainersAndAdjustAllocationEntries(allocationInfosJustFinishRampUp)
-		if err != nil {
-			// not influencing return response to kubelet when putContainersAndAdjustAllocationEntries failed
-			klog.Errorf("[CPUDynamicPolicy.GetResourcesAllocation] putContainersAndAdjustAllocationEntries failed with error: %v", err)
+		if err = p.setAllocationsAndAdjustAllocationEntries(allocationInfosJustFinishRampUp); err != nil {
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:442]: remain

[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:442]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 6]
```diff
@@ -545,16 +548,16 @@ func (p *DynamicPolicy) GetTopologyHints(ctx context.Context,
 	if err != nil {
 		err = fmt.Errorf("GetKatalystQoSLevelFromResourceReq for pod: %s/%s, container: %s failed with error: %v",
 			req.PodNamespace, req.PodName, req.ContainerName, err)
-		klog.Errorf("[CPUDynamicPolicy.GetTopologyHints] %s", err.Error())
+		general.Errorf("%s", err.Error())
 		return nil, err
 	}
 
-	reqInt, err := getReqQuantityFromResourceReq(req)
+	reqInt, err := util.GetQuantityFromResourceReq(req)
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:555]: put to common util. every plugin calls it

[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:555]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 7]
```diff
@@ -598,25 +600,26 @@ func (p *DynamicPolicy) GetResourcePluginOptions(context.Context, *pluginapi.Emp
...
 	if err != nil {
 		err = fmt.Errorf("GetKatalystQoSLevelFromResourceReq for pod: %s/%s, container: %s failed with error: %v",
 			req.PodNamespace, req.PodName, req.ContainerName, err)
-		klog.Errorf("[CPUDynamicPolicy.Allocate] %s", err.Error())
+		general.Errorf("%s", err.Error())
 		return nil, err
 	}
 
-	reqInt, err := getReqQuantityFromResourceReq(req)
+	reqInt, err := util.GetQuantityFromResourceReq(req)
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:617]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 8]
```diff
@@ -710,293 +714,194 @@ func (p *DynamicPolicy) Allocate(ctx context.Context, req *pluginapi.ResourceReq
...
+
+	p.advisorClient = advisorapi.NewCPUAdvisorClient(cpuAdvisorConn)
+	p.advisorConn = cpuAdvisorConn
+	return nil
+}
+
 // cleanPools is used to clean pools-related data in local state
 func (p *DynamicPolicy) cleanPools() error {
-	requestPools := make(map[string]bool)
+	specifiedPools := make(map[string]bool)
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:813]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 9]
```diff
@@ -710,293 +714,194 @@ func (p *DynamicPolicy) Allocate(ctx context.Context, req *pluginapi.ResourceReq
...
-			ownerPool := allocationInfo.OwnerPoolName
-			if requestPoolName != "" {
-				requestPools[requestPoolName] = true
-			} else if ownerPool != "" {
-				requestPools[ownerPool] = true
-			} else if allocationInfo.QoSLevel == consts.PodAnnotationQoSLevelReclaimedCores {
-				requestPools[state.PoolNameReclaim] = true
-			} else if allocationInfo.QoSLevel == consts.PodAnnotationQoSLevelSharedCores {
-				requestPools[state.PoolNameShare] = true
+			ownerPool := allocationInfo.GetOwnerPoolName()
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:823]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 10]
```diff
@@ -1012,6 +917,7 @@ func (p *DynamicPolicy) initReclaimPool() error {
 		}
 
 		// for residual pools, we must make them exist even if cause overlap
+		// todo: residentCPUs is the same as reservedCPUs, why should we do this?
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:920]: +1  not done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 11]
```diff
@@ -1037,1506 +941,40 @@ func (p *DynamicPolicy) initReclaimPool() error {
...
-			cset, availableCPUs, err = calculator.TakeByNUMABalance(p.machineInfo, availableCPUs, quantity)
-			if err != nil {
-				return nil, clonedAvailableCPUs, fmt.Errorf("take cpu for pod: %s container: %s of req: %d failed with error: %v",
-					podUID, containerName, quantity, err)
-			}
-
-			containersCPUSet[podUID][containerName] = cset
+	if allocationInfo.RequestQuantity == 0 {
+		if p.metaServer == nil {
+			klog.Errorf("%v nil metaServer")
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:964]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 12]
```diff
@@ -0,0 +1,585 @@
...
+			if allocationInfo == nil {
+				continue
+			}
+
+			if chkEntries[uid].Entries == nil {
+				chkEntries[uid].Entries = make(map[string]*advisorapi.AllocationInfo)
+			}
+
+			chkEntries[uid].Entries[entryName] = &advisorapi.AllocationInfo{
+				OwnerPoolName: allocationInfo.OwnerPoolName,
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:127]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 13]
```diff
@@ -0,0 +1,585 @@
...
+		// todo, even validation already guarantees that calculationInfo of static pool
+		//  only has one block isn't topology aware, we should not hardcode like this
+		blocks, _ := resp.GetBlock(poolName, advisorapi.FakedContainerID, advisorapi.FakedNumaID)
+		blockID := blocks[0].BlockId
+
+		blockCPUSet[blockID] = allocationInfo.AllocationResult.Clone()
+		availableCPUs = availableCPUs.Difference(blockCPUSet[blockID])
+	}
+
+	// walk through all blocks (that belongs to container)
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:286]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 14]
```diff
@@ -0,0 +1,585 @@
...
+		availableCPUs = availableCPUs.Difference(cpuset)
+	}
+
+	return blockCPUSet, nil
+}
+
+// applyBlocks allocate based on BlockCPUSet
+// and the logic contains three main steps
+// 1. construct entries for dedicated containers and none shared and reclaimed pools
+// 2. construct entries for reclaimed pools
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:354]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 15]
```diff
@@ -0,0 +1,585 @@
...
+// 3. construct entries for shared and reclaimed containers
+// todo but why we don't need to handle shared pools here
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:356]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 16]
```diff
@@ -0,0 +1,585 @@
...
+	if resp == nil {
+		return fmt.Errorf("applyBlocks got nil resp")
+	}
+
+	curEntries := p.state.GetPodEntries()
+	newEntries := make(state.PodEntries)
+	dedicatedCPUSet := machine.NewCPUSet()
+	pooledUnionDedicatedCPUSet := machine.NewCPUSet()
+
+	// deal with blocks of dedicated_cores and (none shared and reclaimed) pools
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:367]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 17]
```diff
@@ -0,0 +1,585 @@
...
+	for entryName, entry := range resp.Entries {
+		for subEntryName, calculationInfo := range entry.Entries {
+			if calculationInfo == nil {
+				general.Warningf("got nil calculationInfo entry: %s, subEntry: %s", entryName, subEntryName)
+				continue
+			} else if !(subEntryName == advisorapi.FakedContainerID || calculationInfo.OwnerPoolName == state.PoolNameDedicated) {
+				continue
+			}
+
+			// construct cpuset for this container by union all blocks for it
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:377]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 18]
```diff
@@ -0,0 +1,585 @@
...
+			}
+
+			// transform cpuset into topologyAwareAssignments
+			topologyAwareAssignments, err := machine.GetNumaAwareAssignments(p.machineInfo.CPUTopology, entryCPUSet)
+			if err != nil {
+				return fmt.Errorf("unable to calculate topologyAwareAssignments for entry: %s, subEntry: %s, entry cpuset: %s, error: %v",
+					entryName, subEntryName, entryCPUSet.String(), err)
+			}
+
+			// if allocation already exits, update them; otherwise, construct new a new one
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:390]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 19]
```diff
@@ -0,0 +1,585 @@
...
+				allocationInfo.OriginalTopologyAwareAssignments = machine.DeepcopyCPUAssignment(topologyAwareAssignments)
+			}
+
+			if newEntries[entryName] == nil {
+				newEntries[entryName] = make(state.ContainerEntries)
+			}
+			newEntries[entryName][subEntryName] = allocationInfo
+			pooledUnionDedicatedCPUSet = pooledUnionDedicatedCPUSet.Union(allocationInfo.AllocationResult)
+
+			// ramp-up finishes immediate for dedicated
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:425]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go [hunk 20]
```diff
@@ -0,0 +1,585 @@
...
+	}
+
+	rampUpCPUs := p.machineInfo.CPUDetails.CPUs().Difference(p.reservedCPUs).Difference(dedicatedCPUSet)
+	rampUpCPUsTopologyAwareAssignments, err := machine.GetNumaAwareAssignments(p.machineInfo.CPUTopology, rampUpCPUs)
+	if err != nil {
+		return fmt.Errorf("unable to calculate topologyAwareAssignments for rampUpCPUs, result cpuset: %s, error: %v",
+			rampUpCPUs.String(), err)
+	}
+
+	// construct entries for reclaimed
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:450]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 21]
```diff
@@ -0,0 +1,1072 @@
...
+	"github.com/kubewharf/katalyst-core/pkg/util/machine"
+)
+
+func (p *DynamicPolicy) sharedCoresAllocationHandler(_ context.Context,
+	req *pluginapi.ResourceRequest) (*pluginapi.ResourceAllocationResponse, error) {
+	if req == nil {
+		return nil, fmt.Errorf("sharedCoresAllocationHandler got nil request")
+	}
+
+	reqInt, err := util.GetQuantityFromResourceReq(req)
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:43]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 22]
```diff
@@ -0,0 +1,1072 @@
...
+		general.Infof("pod: %s/%s, container: %s is still in ramp up, allocate pooled cpus: %s",
+			req.PodNamespace, req.PodName, req.ContainerName, pooledCPUs.String())
+
+		allocationInfo.AllocationResult = pooledCPUs
+		allocationInfo.OriginalAllocationResult = pooledCPUs.Clone()
+		allocationInfo.TopologyAwareAssignments = pooledCPUsTopologyAwareAssignments
+		allocationInfo.OriginalTopologyAwareAssignments = machine.DeepcopyCPUAssignment(pooledCPUsTopologyAwareAssignments)
+	} else {
+		// need to adjust pools and setAllocationsAndAdjustAllocationEntries will set the allocationInfo after adjusted
+		err = p.setAllocationsAndAdjustAllocationEntries([]*state.AllocationInfo{allocationInfo})
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:100]: same to above + 1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 23]
```diff
@@ -0,0 +1,1072 @@
...
+		updatedMachineState, err := state.GenerateMachineStateFromPodEntries(p.machineInfo.CPUTopology, podEntries)
+		if err != nil {
+			general.Errorf("pod: %s/%s, container: %s GenerateMachineStateFromPodEntries failed with error: %v",
+				req.PodNamespace, req.PodName, req.ContainerName, err)
+			return nil, fmt.Errorf("GenerateMachineStateFromPodEntries failed with error: %v", err)
+		}
+		p.state.SetMachineState(updatedMachineState)
+	}
+
+	resp, err := rePackAllocations(allocationInfo, string(v1.ResourceCPU), util.OCIPropertyNameCPUSetCPUs, false, true, req)
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:132]: advise remain +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 24]
```diff
@@ -0,0 +1,1072 @@
...
+
+	err = p.cleanPools()
+	if err != nil {
+		return fmt.Errorf("cleanPools failed with error: %v", err)
+	}
+
+	return nil
+}
+
+// reclaimOverlapNUMABinding unions intersection of current reclaim pool and non-ramp-up dedicated_cores numa_binding containers
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:553]: +1 @csfldf modify comments

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 25]
```diff
@@ -0,0 +1,1072 @@
...
+	unionDedicatedIsolatedCPUSet := machine.NewCPUSet()
+
+	// walk through all isolated CPUSet map to store those pods/containers in pod entries
+	for podUID, containerEntries := range isolatedCPUSet {
+		for containerName, isolatedCPUs := range containerEntries {
+			allocationInfo := curEntries[podUID][containerName]
+			if allocationInfo == nil {
+				general.Errorf("isolated pod: %s, container: %s without entry in current checkpoint", podUID, containerName)
+				continue
+			} else if !state.CheckNumaBinding(allocationInfo) {
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:611]: +1 @csfldf modify it; add comments

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 26]
```diff
@@ -0,0 +1,1072 @@
...
+			containersCPUSet[podUID][containerName] = cset
+		}
+	}
+	return containersCPUSet, availableCPUs, nil
+}
+
+// rePackAllocations regenerates allocations for container that'd already been allocated cpu,
+// and rePackAllocations will assemble allocations based on already-existed AllocationInfo,
+// without any calculation logics at all
+func rePackAllocations(allocationInfo *state.AllocationInfo, resourceName, ociPropertyName string,
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:1031]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/state.go [hunk 27]
```diff
@@ -142,63 +230,81 @@ func (pe PodEntries) String() string {
...
 	if err != nil {
 		klog.Errorf("[PodEntries.String] marshal PodEntries failed with error: %v", err)
 		return ""
 	}
-
 	return string(contentBytes)
 }
 
-func (pe PodEntries) GetPoolCPUset(poolName string) (machine.CPUSet, error) {
+// CheckPoolEmpty returns true if the given pool doesn't exit
```
[the remark this answers is not in the document]
Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/state.go:240]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 28]
```diff
@@ -710,293 +714,194 @@ func (p *DynamicPolicy) Allocate(ctx context.Context, req *pluginapi.ResourceReq
...
-			if allocationInfo == nil {
-				continue
-			}
-
-			if chkEntries[uid].Entries == nil {
-				chkEntries[uid].Entries = make(map[string]*advisorapi.AllocationInfo)
-			}
-
-			chkEntries[uid].Entries[entryName] = &advisorapi.AllocationInfo{
-				OwnerPoolName: allocationInfo.OwnerPoolName,
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:751]: +1

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy.go (hunk 3 above)
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/memory/dynamicpolicy/policy.go:241]: +1

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/cpuadvisor/helper.go [hunk 29]
```diff
@@ -105,6 +105,155 @@ func (m *ListAndWatchResponse) GetBlocks() (map[int]map[string]*Block, error) {
 			}
 		}
 	}
-
 	return blocks, nil
 }
+
+// GetBlock returns Block lists according to the given [entry, subEntry] pair
+func (lwr *ListAndWatchResponse) GetBlock(entry, subEntry string, numa int64) ([]*Block, bool) {
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/cpuadvisor/helper.go:112]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/cpuadvisor/helper.go [hunk 30]
```diff
@@ -17,45 +17,46 @@ limitations under the License.
...
+	"github.com/kubewharf/katalyst-core/pkg/util/machine"
 )
 
 // FakedContainerID represents a placeholder since pool entry has no container-level
 // FakedNumaID represents a placeholder since pools like shared/reclaimed will not contain a specific numa
 const (
-	FakedContainerID = ""
-	FakedNumaID      = -1
+	EmptyOwnerPoolName = ""
+	FakedContainerID   = ""
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/cpuadvisor/helper.go:32]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 31]
```diff
@@ -252,73 +264,67 @@ func (p *DynamicPolicy) Start() (err error) {
...
+	general.Infof("start dynamic policy cpu plugin with sys-advisor")
+	err = p.initAdvisor()
 	if err != nil {
-		klog.Errorf("[CPUDynamicPolicy.Start] initialize failed with error: %v", err)
+		general.Errorf("initAdvisor failed with error: %v", err)
 		return
 	}
 
-	go wait.BackoffUntil(func() { p.serveCPUPluginCheckpoint(p.stopCh) }, wait.NewExponentialBackoffManager(
+	go wait.BackoffUntil(func() { p.serveCPUAdvisor(p.stopCh) }, wait.NewExponentialBackoffManager(
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:298]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 32]
```diff
@@ -252,73 +264,67 @@ func (p *DynamicPolicy) Start() (err error) {
...
 	} else if p.cpuAdvisorSocketAbsPath == "" || p.cpuPluginSocketAbsPath == "" {
 		return fmt.Errorf("invalid cpuAdvisorSocketAbsPath: %s or cpuPluginSocketAbsPath: %s",
 			p.cpuAdvisorSocketAbsPath, p.cpuPluginSocketAbsPath)
 	}
 
-	klog.Infof("[CPUDynamicPolicy.Start] start dynamic policy cpu plugin with sys-advisor")
-
-	err = p.initialize()
+	general.Infof("start dynamic policy cpu plugin with sys-advisor")
+	err = p.initAdvisor()
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:292]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:442]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:617]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:813]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:823]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:964]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:127]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go (hunk 18 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_advisor_handler.go:390]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_hint_handlers.go [hunk 33]
```diff
@@ -93,29 +95,30 @@ func (p *DynamicPolicy) dedicatedCoresWithNUMABindingHintHandler(ctx context.Con
...
-				klog.Errorf("[CPUDynamicPolicy.dedicatedCoresWithNUMABindingHintHandler] pod: %s/%s, container: %s GenerateCPUMachineStateByPodEntries failed with error: %v",
+				general.Errorf("pod: %s/%s, container: %s GenerateMachineStateFromPodEntries failed with error: %v",
 					req.PodNamespace, req.PodName, req.ContainerName, err)
-				return nil, fmt.Errorf("GenerateCPUMachineStateByPodEntries failed with error: %v", err)
+				return nil, fmt.Errorf("GenerateMachineStateFromPodEntries failed with error: %v", err)
 			}
 		}
 	}
 
+	// if hints exit in extra state-file, prefer to use them
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_hint_handlers.go:107]: done

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go [hunk 34]
```diff
@@ -97,27 +98,28 @@ func (p *DynamicPolicy) dedicatedCoresWithNUMABindingHintHandler(ctx context.Con
...
-				klog.Errorf("[MemoryDynamicPolicy.dedicatedCoresWithNUMABindingHintHandler] pod: %s/%s, container: %s GenerateResourcesMachineStateFromPodEntries failed with error: %v",
+				general.Errorf("pod: %s/%s, container: %s GenerateMachineStateFromPodEntries failed with error: %v",
 					req.PodNamespace, req.PodName, req.ContainerName, err)
-				return nil, fmt.Errorf("GenerateResourcesMachineStateFromPodEntries failed with error: %v", err)
+				return nil, fmt.Errorf("GenerateMachineStateFromPodEntries failed with error: %v", err)
 			}
 		}
 	}
 
+	// if hints exit in extra state-file, prefer to use them
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go:110]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/state.go (hunk 27 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/state.go:240]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go (hunk 21 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:43]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go (hunk 22 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:100]: done

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy.go (hunk 3 above)
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/memory/dynamicpolicy/policy.go:241]: done

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_test.go (hunk 1 above)
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_test.go:48]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_hint_handlers.go [hunk 35]
```diff
@@ -62,29 +63,30 @@ func (p *DynamicPolicy) dedicatedCoresHintHandler(ctx context.Context,
...
 
-	var hints map[string]*pluginapi.ListOfTopologyHints
-	allocationInfo := p.state.GetAllocationInfo(req.PodUid, req.ContainerName)
 	machineState := p.state.GetMachineState()
+	var hints map[string]*pluginapi.ListOfTopologyHints
 
+	allocationInfo := p.state.GetAllocationInfo(req.PodUid, req.ContainerName)
 	if allocationInfo != nil {
-		hints = regenerateHints(allocationInfo, reqInt)
+		hints = rePackHints(allocationInfo, reqInt)
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_hint_handlers.go:87]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_async_handler.go [hunk 36]
```diff
@@ -0,0 +1,245 @@
...
+	podEntries := p.state.GetPodEntries()
+	actualCPUSets := make(map[string]map[string]machine.CPUSet)
+	for podUID, containerEntries := range podEntries {
+		if containerEntries.IsPoolEntry() {
+			continue
+		}
+
+		for containerName, allocationInfo := range containerEntries {
+			if allocationInfo == nil ||
+				allocationInfo.ContainerType != pluginapi.ContainerType_MAIN.String() {
```
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_async_handler.go:51]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go (hunk 26 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:1031]: done

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go (hunk 4 above)
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_hint_handlers.go:101]: done

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (waynepeking348) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go]: done

**Review by csfldf — APPROVED** [review]

**Review by luomingmeng — APPROVED** [review]

**Review by luomingmeng — APPROVED** [review]

**Review by luomingmeng — APPROVED** [review]

**Pull request merged**